images/hooks/pkg/hooks/create-generic-vmclass/hook.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2025 Flant JSC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create_generic_vmclass
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"hooks/pkg/settings"
+
+	"github.com/deckhouse/virtualization/api/core"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2"
+
+	"github.com/deckhouse/module-sdk/pkg"
+	"github.com/deckhouse/module-sdk/pkg/registry"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	moduleStateSecretSnapshot = "module-state-secret"
+	vmClassSnapshot           = "vmclass-generic"
+
+	moduleStateSecretName = "module-state"
+	genericVMClassName    = "generic"
+
+	apiVersion = core.GroupName + "/" + v1alpha2.Version
+)
+
+var _ = registry.RegisterFunc(config, Reconcile)
+
+var config = &pkg.HookConfig{
+	OnBeforeHelm: &pkg.OrderedConfig{Order: 5},
+	Kubernetes: []pkg.KubernetesConfig{
+		{
+			Name:       moduleStateSecretSnapshot,
+			APIVersion: "v1",
+			Kind:       "Secret",
+			JqFilter:   `.data`,
+			NameSelector: &pkg.NameSelector{
+				MatchNames: []string{moduleStateSecretName},
+			},
+			NamespaceSelector: &pkg.NamespaceSelector{
+				NameSelector: &pkg.NameSelector{
+					MatchNames: []string{settings.ModuleNamespace},
+				},
+			},
+			ExecuteHookOnSynchronization: ptr.To(false),
+		},
+		{
+			Name:       vmClassSnapshot,
+			APIVersion: apiVersion,
+			Kind:       v1alpha2.VirtualMachineClassKind,
+			JqFilter:   `.metadata.name`,
+			NameSelector: &pkg.NameSelector{
+				MatchNames: []string{genericVMClassName},
+			},
+			LabelSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app":    "virtualization-controller",
+					"module": settings.ModuleName,
+				},
+			},
+			ExecuteHookOnSynchronization: ptr.To(false),
+		},
+	},
+
+	Queue: fmt.Sprintf("modules/%s", settings.ModuleName),
+}
+
+func Reconcile(_ context.Context, input *pkg.HookInput) error {
+	moduleStateSecrets := input.Snapshots.Get(moduleStateSecretSnapshot)
+	vmClasses := input.Snapshots.Get(vmClassSnapshot)
+
+	// nothing to do if generic vmclass already exists
+	if len(vmClasses) > 0 {
+		return nil
+	}
+
+	// if module-state secret exists and contains generic-vmclass-created=true, nothing to do
```

Comment:
Strange name for the flag. vmclass created, but there is no vmclass =) May be something like "vmclass-generic-was-initiated"?

Reply:
Done:  vmClassGeneric field for the secret and "installedAt: "datetime" json object inside.